plugins/formal-verification/src/org/jetbrains/kotlin/formver/conversion/ContractDescriptionConversionVisitor.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.conversion
+
+import org.jetbrains.kotlin.contracts.description.*
+import org.jetbrains.kotlin.fir.contracts.description.ConeContractConstantValues
+import org.jetbrains.kotlin.fir.diagnostics.ConeDiagnostic
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+import org.jetbrains.kotlin.formver.scala.silicon.ast.Exp
+import org.jetbrains.kotlin.formver.scala.silicon.ast.Type
+
+class ContractDescriptionConversionVisitor : KtContractDescriptionVisitor<Exp, StmtConversionContext, ConeKotlinType, ConeDiagnostic>() {
+    override fun visitConstantDescriptor(
+        constantReference: KtConstantReference<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        val retType = data.methodCtx.returnVar.type.viperType
+        return when (constantReference) {
+            ConeContractConstantValues.WILDCARD -> Exp.BoolLit(true)
+            ConeContractConstantValues.NULL -> Exp.EqCmp(Exp.LocalVar("ret\$", retType), Exp.NullLit())
+            ConeContractConstantValues.NOT_NULL -> Exp.NeCmp(Exp.LocalVar("ret\$", retType), Exp.NullLit())
+            ConeContractConstantValues.TRUE -> Exp.EqCmp(Exp.LocalVar("ret\$", retType), Exp.BoolLit(true))
+            ConeContractConstantValues.FALSE -> Exp.EqCmp(Exp.LocalVar("ret\$", retType), Exp.BoolLit(false))
+            else -> throw Exception("Unexpected constant: $constantReference")
+        }
+    }
+
+    override fun visitReturnsEffectDeclaration(
+        returnsEffect: KtReturnsEffectDeclaration<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        return returnsEffect.value.accept(this, data)
+    }
+
+    override fun visitBooleanValueParameterReference(
+        booleanValueParameterReference: KtBooleanValueParameterReference<ConeKotlinType, ConeDiagnostic>,
+        data: StmtConversionContext
+    ): Exp {
+        // TODO: find a better way to do that
+        return Exp.LocalVar("local\$${booleanValueParameterReference.name}", Type.Bool)
```

Comment:
in particular, have a look at this because I'm sure that it is not correct (it works just for the tests) but I don't know what is the best way to get the var

Reply:
It's a bit hacky, but I think `LocalName` should work for this.   You might want to make an extension method for it.